darts/models/forecasting/forecasting_model.py

```diff
@@ -934,52 +1303,49 @@ def residuals(
             Whether to print progress.
         Returns
         -------
-        TimeSeries
+        TimeSeries (or Sequence[TimeSeries])
             The vector of residuals.
         """
-        try:
-            series._assert_univariate()
-        except (AttributeError, TypeError):
-            raise ValueError(
-                "series must be of type TimeSeries. "
-                "If Sequence[TimeSeries] is provided, select the series to compute residuals for."
-            )
-
-        if past_covariates is not None:
-            raise_if_not(
-                isinstance(past_covariates, TimeSeries),
-                "past_covariates should be of type TimeSeries",
-            )
 
-        if future_covariates is not None:
-            raise_if_not(
-                isinstance(future_covariates, TimeSeries),
-                "future_covariates should be of type TimeSeries",
-            )
+        series = series2seq(series)
+        past_covariates = series2seq(past_covariates)
+        future_covariates = series2seq(future_covariates)
 
-        # get first index not contained in the first training set
-        first_index = series.time_index[self.min_train_series_length]
-
-        # compute fitted values
-        p = self.historical_forecasts(
-            series=series,
-            past_covariates=past_covariates,
-            future_covariates=future_covariates,
-            start=first_index,
-            forecast_horizon=forecast_horizon,
-            stride=1,
-            retrain=retrain,
-            last_points_only=True,
-            verbose=verbose,
+        raise_if_not(
+            all([serie.is_univariate for serie in series]),
+            "Each series in the sequence must be univariate.",
+            logger,
         )
 
+        residuals_list = []
         # compute residuals
-        series_trimmed = series.slice_intersect(p)
-        residuals = series_trimmed - (
-            p.quantile_timeseries(quantile=0.5) if p.is_stochastic else p
-        )
+        for idx, target_ts in enumerate(series):
```

Comment:
Here too a generator might be cleaner and more scalable. If we go this way, we could change the check above (`all([serie.is_univariate for serie in series])` and check each series one-by-one inside the loop, instead of having the eager version.